app/controllers/signup/signup-ctrl.js

```diff
@@ -1,43 +1,24 @@
 var app = angular.module('myApp');
 
-app.controller('SignUpController', ['$scope', '$log', '$location', function($scope, $log, $location) {
+app.controller('SignUpController', ['$scope', '$log', '$location', 'UserService',
+    function($scope, $log, $location, userService) {
 
     $scope.showConfirm = false;
-
     $scope.showUserNameTaken = false;
 
     $scope.signupForm = function() {
 
         if ($scope.signup_form.$valid) {
-
-            var user = new Parse.User();
-
-            user.set("username", $scope.signup.username);
-            user.set("password", $scope.signup.password);
-            user.set("email", $scope.signup.email);
-
-            user.signUp(null, {
-                success: function(object) {
-                    $log.info('User saved');
-                    $scope.$apply($scope.newUserAccountCreated());
-                },
-                error: function(model, error) {
-                    $log.info('User not saved with error' + error.message);
-                    $scope.$apply($scope.errorInNewAccount());
-                }
-            });
+            userService.createUser($scope.signup.username, $scope.signup.password, $scope.signup.email)
+                .then(function (createdUser) {
+                    $scope.showConfirm = true;
+                    $location.path('/signIn');
+                }).catch(function (error) {
```

Comment:
I get an error here on L#16 that undefined is not a function and I am not able to figure out why